File: tests/machine/x/php/save_jsonl_stdout.php
<?php
// people: [{string: any}]
$people = [["name" => "Alice", "age" => 30], ["name" => "Bob", "age" => 25]];
_save_json($people, "-");

function _save_json($rows, $path) {
    $out = json_encode($rows);
    if ($path === '' || $path === '-') { fwrite(STDOUT, $out . PHP_EOL); } else { file_put_contents($path, $out); }
}

File: tests/machine/x/ocaml/save_jsonl_stdout.ml
let save_jsonl rows path =
  let oc = if path = "-" then stdout else open_out path in
  List.iter (fun m ->
    let parts = List.map (fun (k,v) -> Printf.sprintf "\"%s\": %s" k (__show (Obj.obj v))) m in
    output_string oc ("{" ^ String.concat ", " parts ^ "}\n")
  ) rows;
  if path <> "-" then close_out oc


let people = [[("name",Obj.repr ("Alice"));("age",Obj.repr (30))];[("name",Obj.repr ("Bob"));("age",Obj.repr (25))]]

let () =
  save_jsonl people "-";
